jose/jwe.py

```diff
@@ -76,6 +76,13 @@ def decrypt(jwe_str, key):
         >>> jwe.decrypt(jwe_string, 'asecret128bitkey')
         'Hello, World!'
     """
+
+    # Limit the token size - if the data is compressed then decompressing the
+    # data could lead to large memory usage. This helps address This addresses
+    # CVE-2024-33664. Also see _decompress()
+    if len(jwe_str) > JWE_SIZE_LIMIT:
+        raise JWEError("JWE string exceeds {JWE_SIZE_LIMIT} bytes")
```

Comment:
```suggestion
        raise JWEError(f"JWE string exceeds {JWE_SIZE_LIMIT} bytes")
```

Reply:
@alistairwatts Could you please accept this change?